oracle/src/oracleimporter.cpp

```diff
@@ -318,6 +322,26 @@ int OracleImporter::importTextSpoiler(CardSetPtr set, const QVariant &data)
     return cards;
 }
 
+void OracleImporter::sortColors(QStringList &colors)
+{
+    QMap<QString, unsigned int> colorOrder;
+    colorOrder["W"] = 0;
+    colorOrder["U"] = 1;
+    colorOrder["B"] = 2;
+    colorOrder["R"] = 3;
+    colorOrder["G"] = 4;
+    bool correctOrder = false;
+    while (!correctOrder) {
+        correctOrder = true;
+        for (unsigned int i = 0; i < colors.length() - 1; i++) {
+            if (colorOrder[colors[i]] > colorOrder[colors[i + 1]]) {
```

Comment:
If i add a new fake color to the xml file (eg: `Z`), this is going to crash.

Reply:
That shouldn't be a problem, because this whole sorting happens before the xml is created, during the mtgjson processing. The only input we'll ever get in here is the output of the `OracleImporter::extractColors` method, which is guaranteed to be one of the five valid color codes.